### ISSUE TITLE
Interaction matrix for a non-modulated PWFS

### ISSUE THREAD

**Issue by jnousi** [ANR-COMPASS/shesha#36]
Hi,

If I understood the code correctly, the centroiding algorithms for the pyramid sensor scale the measurements with respect to modulation amplitude. Consequently, when you take modulations amplitude to zero, you get all zeros interaction matrix.

(It seems that) I can bypass the problem by manipulating the shesha code (or taking the modulations amplitude very close to 0), which is not the most elegant solution. I wonder if there's a "right way" to run non-modulated PWFS?

Many thanks,
Jalo

**Comment by jnousi** [comment]
Thanks, Arnaud

**Comment by a-sevin** [comment]
No problem, just a comment. 

Because of the complexity of building a perfect pyramid, the extreme sensitivity (which leads to saturation of the measurement) and the non-linearity of the non-modulated PWFS, I advise to use a modulated pyramid.
Moreover, with the non-modulated PWFS, there are a lot of diffraction effects made by the edges of the pyramid which produce signals between the 4 pupils of the pyramid and the control law is very difficult.
In practice, I do not know of any AO bench working with non-modulated PWFS (but few benches are using a tiny modulation amplitude)

**Comment by jnousi** [comment]
Yep, thanks. We just wanted to study the robustness of ML methods to non-linearity in the control loop. Actually, with relatively easy conditions, it seems easy to close the loop with the integrator also.
